src/elements/common/item/iconCellRenderer.js

```diff
@@ -5,28 +5,52 @@
  */
 
 import React from 'react';
+import { injectIntl } from 'react-intl';
+import type { IntlShape } from 'react-intl';
+
 import FileIcon from '../../../icons/file-icon/FileIcon';
 import FolderIcon from '../../../icons/folder-icon/FolderIcon';
 import BookmarkIcon from '../../../icons/bookmark-icon/BookmarkIcon';
-import { TYPE_FOLDER, TYPE_FILE, TYPE_WEBLINK } from '../../../constants';
 import type { BoxItem } from '../../../common/types/core';
 
+import { TYPE_FOLDER, TYPE_FILE, TYPE_WEBLINK } from '../../../constants';
+import messages from '../messages';
+
 import './IconCell.scss';
 
-export function getIcon(dimension: number, rowData: BoxItem) {
+export function getIcon(intl: IntlShape, rowData: BoxItem, dimension?: number) {
     const { type, extension, has_collaborations, is_externally_owned }: BoxItem = rowData;
+    let title;
     switch (type) {
-        case TYPE_FOLDER:
-            return <FolderIcon dimension={dimension} isCollab={has_collaborations} isExternal={is_externally_owned} />;
         case TYPE_FILE:
-            return <FileIcon dimension={dimension} extension={extension} />;
+            title = intl.formatMessage(messages.file);
+            return <FileIcon dimension={dimension} extension={extension} title={title} />;
+        case TYPE_FOLDER:
+            if (has_collaborations) {
+                title = intl.formatMessage(messages.collaboratedFolder);
+            } else if (is_externally_owned) {
+                title = intl.formatMessage(messages.externalFolder);
+            } else {
+                title = intl.formatMessage(messages.personalFolder);
+            }
+            return (
+                <FolderIcon
+                    dimension={dimension}
+                    isCollab={has_collaborations}
+                    isExternal={is_externally_owned}
+                    title={title}
+                />
+            );
         case TYPE_WEBLINK:
-            return <BookmarkIcon height={dimension} width={dimension} />;
+            title = intl.formatMessage(messages.bookmark);
+            return <BookmarkIcon height={dimension} width={dimension} title={title} />;
         default:
-            throw new Error('Unsupported item type!');
+            title = intl.formatMessage(messages.file);
+            return <FileIcon dimension={dimension} title={title} />;
     }
 }
 
-export default (dimension: number = 32): Function => ({ rowData }: { rowData: BoxItem }) => (
-    <div className="be-item-icon">{getIcon(dimension, rowData)}</div>
-);
+export default (dimension: number = 32) =>
+    injectIntl(({ intl, rowData }: { intl: IntlShape, rowData: BoxItem }) => (
+        <div className="be-item-icon">{getIcon(intl, rowData, dimension)}</div>
```

Comment:
Is it possible to inject getIcon with intl?

Reply:
I think so. I turned `getIcon` into an explicit component (renamed to `IconCell`) instead of a function.

There is a slight difference, which is that the `ItemListIcon` component hierarchy now includes the `IconCell` component. So in the unit tests, `shallow(ItemListicon)` would see `IconCell` instead of `FileIcon`/`FolderIcon`/`BookmarkIcon`.